app/subscriber/src/features/commentary/Commentary.tsx

```diff
@@ -1,68 +1,119 @@
-import { DetermineContentIcon, isWeekday } from './utils';
+import {
+  castToSearchResult as castMessageToSearchResult,
+  filterCommentaryResults,
+  getCutoff,
+} from './utils';
 
+/**
+ * Provides a section on the page to display the current commentary stories.
+ * This displays stories that include the commentary action and are within 3 hours weekdays or 5 on weekends.
+ * @returns Component
+ */
 export const Commentary: React.FC = () => {
   const [, { findContentWithElasticsearch }] = useContent();
-  const [commentary, setCommentary] = React.useState<IContentSearchResult[]>();
+  const [commentary, setCommentary] = React.useState<IContentSearchResult[]>([]);
   const navigateAndScroll = useNavigateAndScroll();
   const { commentaryActionId, isReady } = useSettings();
   const getActionFilters = useActionFilters();
+  const hub = useApiHub();
 
-  /** determine how far back to grab commentary */
-  const determineCommentaryTime = () => {
-    const date = new Date();
-    if (isWeekday()) {
-      date.setHours(date.getHours() - 3);
-    } else {
-      date.setHours(date.getHours() - 5);
-    }
-    return moment(date).toISOString();
-  };
-
-  React.useEffect(() => {
-    if (isReady) {
+  const fetchCommentary = React.useCallback(
+    async (commentaryActionId: number) => {
       let actionFilters = getActionFilters();
       const commentaryAction = actionFilters.find((a) => a.id === commentaryActionId);
 
-      findContentWithElasticsearch(
-        generateQuery(
-          filterFormat({
-            actions: commentaryAction ? [commentaryAction] : [],
-            searchUnpublished: false,
-            startDate: determineCommentaryTime(),
-            size: 100,
+      const query = generateQuery(
+        filterFormat({
+          actions: commentaryAction ? [commentaryAction] : [],
+          searchUnpublished: false,
+          startPostedDate: getCutoff(),
+          size: 100,
+        }),
+      );
+
+      const res = await findContentWithElasticsearch(query, false);
+      setCommentary(
+        filterCommentaryResults(
+          res.hits.hits.map((r) => {
+            const content = r._source as IContentModel;
+            return castToSearchResult(content);
           }),
+          commentaryActionId,
         ),
-        false,
-      )
-        .then((res) => {
-          setCommentary(
-            res.hits.hits.map((r) => {
-              const content = r._source as IContentModel;
-              return castToSearchResult(content);
-            }),
-          );
-        })
-        .catch(() => {});
+      );
+    },
+    [findContentWithElasticsearch, getActionFilters],
+  );
+
+  React.useEffect(() => {
+    if (isReady && commentaryActionId) {
+      fetchCommentary(commentaryActionId).catch(() => {});
     }
-  }, [commentaryActionId, findContentWithElasticsearch, getActionFilters, isReady]);
+  }, [commentaryActionId, fetchCommentary, isReady]);
+
+  const onContentReceived = React.useCallback(
```

Comment:
We now listen for SignalR messages for content.  This keeps the Commentary section up-to-date.